tensordict/prototype/tensorclass.py

```diff
@@ -123,14 +148,34 @@ def __torch_function__(
         args: tuple[Any, ...] = (),
         kwargs: dict[str, Any] | None = None,
     ) -> Callable:
+        if (
+            func not in tc_handled_functions and func not in _TD_PASS_THROUGH
+        ) or not all(issubclass(t, (Tensor, cls)) for t in types):
+            return NotImplemented
         if kwargs is None:
             kwargs = {}
-        if func not in td_handled_functions or not all(
-            issubclass(t, (Tensor, cls)) for t in types
-        ):
-            return NotImplemented
+        if func in _TD_PASS_THROUGH:
+            if len(args) > 0:
+                input_ = args[0]
+            else:
+                input_ = kwargs["input"] if "input" in kwargs else kwargs["tensors"]
+            if isinstance(input_, (list, tuple)):
+                tc = input_[0]
+                input_ = input_.__class__(tc._tensordict for tc in input_)
+            else:
+                tc = input_
+                input_ = input_._tensordict
 
-        return td_handled_functions[func](*args, **kwargs)
+            out = kwargs.get("out")
+            if out is not None:
+                kwargs["out"] = out._tensordict
```

Comment:
I think this is too brittle, as `out` could be passed as a positional argument in which case this breaks. I wonder if we should instead replace _all_ values in `args` and `kwargs` that happen to be tensorclasses with `value._tensordict`?

Reply:
Yep I guess we could just scan kwargs and replace the tensorclasses by the tensordict they contain

Reply:
I've done this